ls-alg/main/Main.java

```diff
@@ -0,0 +1,162 @@
+package main;
```

Comment:
I don't think we need another folder for `main`. It can all be in the same folder.